routes/cube_routes.js

```diff
@@ -73,6 +74,323 @@ const { ensureAuth, csrfProtection, flashValidationErrors, jsonValidationErrors
 
 router.use(csrfProtection);
 
+const fetchLands = {
+  'Arid Mesa': ['W', 'R'],
+  'Bloodstained Mire': ['B', 'R'],
+  'Flooded Strand': ['W', 'U'],
+  'Marsh Flats': ['W', 'B'],
+  'Misty Rainforest': ['U', 'G'],
+  'Polluted Delta': ['U', 'B'],
+  'Scalding Tarn': ['U', 'R'],
+  'Verdant Catacombs': ['B', 'G'],
+  'Windswept Heath': ['W', 'G'],
+  'Wooded Foothills': ['R', 'G'],
+};
+
+function arrayIsSubset(needles, haystack) {
+  return needles.every((x) => haystack.includes(x));
+}
+
+function botRating(botColors, card, rating) {
+  const colors = card.colors || card.details.color_identity;
+  const subset = arrayIsSubset(colors, botColors) && colors.length > 0;
+  const overlap = botColors.some((c) => colors.includes(c));
+  const typeLine = card.type_line || card.details.type;
+  const isLand = typeLine.indexOf('Land') > -1;
+  const isFetch = fetchLands.includes(card.details.name);
+
+  if (isLand) {
+    if (subset) {
+      // if fetches don't have the color identity override, they get lumped into this category
+      rating *= 1.4;
+    } else if (overlap || isFetch) {
+      rating *= 1.2;
+    } else {
+      rating *= 1.1;
+    }
+  } else if (subset) {
+    rating *= 1.3;
+  } else if (overlap) {
+    rating *= 1.1;
+  }
+
+  return rating;
+}
+
+function convertCard(card) {
+  if (Array.isArray(card)) {
+    return card.map(convertCard);
+  }
+  if (typeof card === 'string' || card instanceof String) {
+    const details = carddb.cardFromId(card);
+    return {
+      tags: [],
+      colors: details.colors,
+      cardID: details._id,
+      cmc: details.cmc,
+      type_line: details.type,
+    };
+  }
+  return card;
+}
+
+// temprorary draft and deck conversion functions
+async function updateDraft(draft) {
+  try {
+    if (draft.seats && draft.seats.length > 0) {
+      return draft;
+    }
+
+    draft.seats = [];
+    draft.unopenedPacks = [];
+
+    // add player
+    const playerSeat = {
+      bot: null,
+      userid: draft.owner,
+      name: draft.username,
+      pickorder: draft.pickOrder ? draft.pickOrder.map(convertCard) : [],
+      drafted: draft.picks[0].map(convertCard),
+      packbacklog: draft.packs[0] && draft.packs[0][0] ? [draft.packs[0][0]] : [],
+    };
+
+    draft.seats.push(playerSeat);
+    draft.unopenedPacks.push(draft.packs[0] ? draft.packs[0].slice(1) : []);
+
+    // add bots
+    for (let i = 1; i < draft.picks.length; i++) {
+      const bot = {
+        bot: draft.bots[i - 1],
+        name: `Bot ${i}: ${draft.bots[i - 1][0]}, ${draft.bots[i - 1][1]}`,
+        pickorder: draft.picks[i].map(convertCard),
+        drafted: [],
+        packbacklog: draft.packs[i] && draft.packs[i][0] ? [draft.packs[i][0]] : [],
+      };
+
+      // now we need to build picks from the pickorder ids
+      for (let j = 0; j < 16; j++) {
+        bot.drafted.push([]);
+      }
+
+      bot.pickorder.forEach((cardid, index) => {
```

Comment:
It doesn't look like you need index so could this be done as a `for... of` loop?